File: src/main/java/fluid/entity/MobileDimensionalEntity.java
package fluid.entity;

/**
 * Created by mjanes on 6/12/2014.
 */
public interface MobileDimensionalEntity extends DimensionalEntity {

    void setDeltaX(double deltaX);

    void addDeltaX(double deltaDeltaX);

    double getDeltaX();

    void moveX(double deltaX);

    void setDeltaY(double deltaY);

    void addDeltaY(double deltaDeltaY);

    double getDeltaY();

    void moveY(double deltaY);

    void setDeltaZ(double deltaZ);

    void addDeltaZ(double deltaDeltaZ);

    double getDeltaZ();

    void moveZ(double deltaZ);

    void move();


}

File: src/main/java/fluid/entity/MobileDimensionalEntity.kt
package fluid.entity

/**
 * Created by mjanes on 6/12/2014.
 */
interface MobileDimensionalEntity : DimensionalEntity {
    fun addDeltaX(deltaDeltaX: Double)
    var deltaX: Double
    fun moveX(deltaX: Double)
    fun addDeltaY(deltaDeltaY: Double)
    var deltaY: Double
    fun moveY(deltaY: Double)
    fun addDeltaZ(deltaDeltaZ: Double)
    var deltaZ: Double
    fun moveZ(deltaZ: Double)
    fun move()
}